### DISCUSSION TITLE
Extracting Bose-Einstein distribution function for SCPH calculations

### DISCUSSION THREAD

**Discussion by eeshaAnd** [ttadano/alamode#98] [Q&A]
Hi everyone,
   I wanted to know if it is possible to extract and plot the Bose-Einstein distribution function as function of temperature for renormalized anharmonic phonon frequencies using the code?

I want to explain a sudden lowering of MSD value at 300K  and in Free energy at around 200K ( Please find attached both the graphs).

Best,
Eesha
![Freeenergy_5c](https://user-images.githubusercontent.com/107148857/195872994-d701315e-503c-412e-8a9c-f7c36f512ac2.jpg)
![MSD_5B](https://user-images.githubusercontent.com/107148857/195873043-ba44cc14-25f1-4733-97d5-59981676fd40.jpg)

**Comment by ttadano** [comment]
The code does not have a function to do that, but it is straight forward to compute the Bose-Einstein occupation function from the temperature and anharmonic phonon frequency.

About the sudden change of the vibrational free energy, I guess the SCPH calculation did not converge in the low-temperature region. Did you check if the SCPH loop reached convergence?

**Comment by eeshaAnd** [comment]
Hi Terumasa,
   It reached convergence but had negative values. PFA the Phonon dispersion graph.
How to improve this? I have changed alpha, convergence threshold and max_iterations in the input file, but the best I could get was this.

Best,
Eesha

________________________________

**Reply by ttadano** [reply]
Do you mean you obtained unstable phonon modes even when the SCPH calculation converged? If so, what was `KMESH_INTERPOLATE` you used and which q points you found imaginary phonons?

**Comment by eeshaAnd** [comment]
Hi Terumasa,
 1) Yes, I have obtained unstable phonons in-spite of convergence.
 2) I have used the following input paramters:
&scph
 SELF_OFFDIAG = 0
 MAXITER = 20000
 MIXALPHA = 0.01
 KMESH_INTERPOLATE = 5 5 1
 KMESH_SCPH = 10 10 1
 TOL_SCPH = 5.0e-8
/
3. The negative q-point values are shown in the graph. Also, I am unable to include the non-analytical corrections. For this one it does not converge.

Thanks, in advance!

Best,
Eesha


________________________________

**Reply by ttadano** [reply]
Where can I find the graph of dispersion curves?

**Reply by eeshaAnd** [reply]
[
![4a-bands-comparison-FINAL](https://user-images.githubusercontent.com/107148857/195930933-79a85140-f6f9-4831-b7e8-4e5b90aabb80.jpg)
![4b-Phonon-DOS-Comparitive](https://user-images.githubusercontent.com/107148857/195930935-1632556d-f533-4795-bbc0-5151280510d8.jpg)
](url)

**Reply by ttadano** [reply]
This result indicates that the ZA mode frequency near the Gamma point became negative after interpolation even when the SCPH calculation reached convergence at 5x5x1 q points. 

I think this can happen for 2D systems, where it is often necessary to use a fairly large supercell along the in-plane direction to remove the imaginary frequency pocket around Gamma point even at the harmonic level.

I would suggest trying the following:

1. Compute harmonic force constants using a large supercell (`M x M x 1`), say 6x6x1 and larger, and confirm that the ZA mode is stable even near the Gamma point.
2. Compute anharmonic force constants using a smaller supercell. I guess 5x5x1 is OK.
3. Use the above two sets of force constants and run a SCPH calculation with `KMESH_INTERPOLATE = N N 1`. Here, `N` is recommended to be 2 or 3, depending on the size of the larger supercell. (`N` must be a divisor of `M`). 

The imaginary frequency pocket around Gamma may be avoided by using a smaller `N`, and this would be a reasonable choice if the anharmonic interaction is shorter-ranged as compared to the harmonic interaction.

The above procedure may not solve the issue. If not, I recommend increasing the temperature and computing free energies only above the temperature where the imaginary mode disappears.

**Comment by eeshaAnd** [comment]
PFA the dispersion graphs.
________________________________
